packages/datadog-plugin-prisma/test/index.spec.js

```diff
@@ -0,0 +1,328 @@
+// from the temporary directory after tests are done
+async function cleanupMockPrismaClient (directory) {
+  const nodeModulesPath = path.join(directory, 'node_modules')
+  const packageJsonPath = path.join(directory, 'package.json')
+  const prismaSchemaPath = path.join(directory, 'schema.prisma')
+
+  await fs.rm(nodeModulesPath, { recursive: true, force: true })
+  await fs.rm(packageJsonPath, { force: true })
+
+  await fs.rm(prismaSchemaPath, { force: true })
+}
+
+describe('Plugin', () => {
+  let prisma
+  let prismaClient
+  let tracingHelper
+
+  describe('prisma', () => {
+    let tempDir
+
+    after(async () => {
+      await cleanupMockPrismaClient(tempDir)
+    })
+    withVersions('prisma', ['@prisma/client'], version => {
+      before(async () => {
+        tempDir = await generatePrismaClient(version)
+      })
+
+      describe('without configuration', () => {
+        before(() => {
+          return agent.load('prisma')
+        })
+
+        beforeEach(() => {
+          prisma = require(`../../../versions/@prisma/client@${version}`).get()
+          prismaClient = new prisma.PrismaClient()
+          tracingHelper = global.PRISMA_INSTRUMENTATION.helper || global.V0_PRISMA_INSTRUMENTATION.helper
+        })
+
+        after(() => { return agent.close({ ritmReset: false }) })
+
+        it('should do automatic instrumentation', async () => {
+          const tracingPromise = agent.assertSomeTraces(traces => {
+            expect(traces[0][0].resource).to.equal('queryRaw')
+            expect(traces[0][0].meta).to.have.property('prisma.type', 'client')
+            expect(traces[0][0].meta).to.have.property('prisma.method', 'queryRaw')
+            expect(traces[0][0]).to.have.property('name', expectedSchema.client.opName)
+            expect(traces[0][0]).to.have.property('service', expectedSchema.client.serviceName)
+
+            // grabbing actual db query span
+            const engineDBSpan = traces[0].find(span => span.meta['prisma.name'] === 'db_query')
+            expect(engineDBSpan).to.have.property('resource', 'SELECT 1')
+            expect(engineDBSpan).to.have.property('type', 'sql')
+            expect(engineDBSpan.meta).to.have.property('span.kind', 'client')
+            expect(engineDBSpan).to.have.property('name', expectedSchema.engine.opName)
+            expect(engineDBSpan).to.have.property('service', expectedSchema.engine.serviceName)
+          })
+
+          await Promise.all([
+            prismaClient.$queryRaw`SELECT 1`,
+            tracingPromise
+          ])
+        })
+
+        it('should handle errors', async () => {
+          let error
+          const tracingPromise = agent.assertSomeTraces(traces => {
+            expect(traces[0][0].meta).to.have.property(ERROR_TYPE, error.name)
+            expect(traces[0][0].meta).to.have.property(ERROR_MESSAGE, error.message)
+            expect(traces[0][0].meta).to.have.property(ERROR_STACK, error.stack)
+          })
```

Comment:
We have a new nicer helper method that is easier to read, use, and it produces better error messages by showing all information at once.

Please also apply this pattern to **all** cases where applicable. That is almost of all these calls.

```suggestion
          const tracingPromise = agent.assertFirstTraceSpan({
            meta: {
              [ERROR_TYPE]: error.name,
              [ERROR_MESSAGE]: error.message,
              [ERROR_STACK]: error.stack
            }
          })
```